javascript/modules/client/src/internal/client-core.ts

```diff
@@ -166,4 +175,37 @@ export abstract class ClientCore implements IClientCore {
       }
     );
   }
+  public async pin(input: string | Uint8Array): Promise<string> {
+    if (!this.ipfs)
+      return Promise.reject(new Error("IPFS client is not initialized"));
+    return this.ipfs
+      .add(input)
+      .then(res => {
+        return Promise.resolve(res.path);
+      })
+      .catch(e => {
+        return Promise.reject(e);
+      });
+  }
+  public async fetchBytes(cid: string) {
+    if (!this.ipfs)
+      return Promise.reject(new Error("IPFS client is not initialized"));
+    try {
+      for await (const chunk of this.ipfs.cat(cid)) {
+        return Promise.resolve(chunk);
+      }
+    } catch (e) {
+      Promise.reject(e);
```

Comment:
We're not returning anything here.
I would personally throw an Error of ourselves